inline-caching/README.md

```diff
@@ -0,0 +1,247 @@
+GemFire's AsyncEventListener and CacheLoader are useful features that can help improve the performance and scalability of your application by allowing you to asynchronously process write events and read data within your GemFire cluster.
+
+### AsyncEventListener
+The AsyncEventListener interface provides a way for your application to send write events to your database asynchronously in a batch operation, without blocking the sender. 
+This can help improve the throughput and performance of your application by allowing it to process events in parallel, while the sender continues to send more write events.
+This also helps keep the data in GemFire in sync with what is in the database.  
+
+In order to simplify the SQL necessary to write data to Postgres, the [JOOQ library](https://www.jooq.org/) is used. 
+
+```java
+int result = create.update(table)
+                    .set(filler, value)
+                    .where(tid.eq(itemId))
+                    .execute();
+```
+
+The table `pgbench_tellers` and column `filler` is something we get when postgres created our benchmark database. 
+It's just a place to store data for our example.
+
+We are getting our postgres credentials from System properties
+```java
+		String userName = System.getProperty("postgres.username");
+		String password = System.getProperty("postgres.password");
+```
+
+These properties will be passed in later when we create our GemFire region below.
+
+
+### CacheLoader
+GemFire's CacheLoader is used to fetch data from external systems and load it into the GemFire cache. 
+Our CacheLoader implementation also uses JOOQ for connecting to Postgres and Java system properties to retrieve Postgres' username and password.
+
+### AsyncEventListener
+Gemfire's AsyncEventListener is used to listen for events to the cache, and then perform a batch update to the backing postgres database.
+HikariCP, a popular connection pooling library for Java applications, is being used here to help handle our postgres connections. We also
+are using a singleton pattern in order to load only one connection pool in our cache. 
+
+### Create a region with an AsyncEventListener and a CacheLoader
+
+To create a region with an AsyncEventListener and a CacheLoader using GemFire, you can use `gfsh`. 
+Be sure that you have already built our example project and downloaded the necessary JDBC drivers as outlined above. 
+We will be creating a GemFire region with the name `item`.
+```bash
+./<location of gemfire>/bin/gfsh
+start locator --name locator
+start server --name='server' --classpath="<project root directory>/cache-loader/build/libs/*" --J="-Dpostgres.username=myuser" --J="-Dpostgres.password=mypass"
+create async-event-queue --listener=io.vmware.event.ItemAsyncEventListener --id=item-writebehind-queue --batch-size=10 --batch-time-interval="20"
+create region --name=item --type=REPLICATE --cache-loader=io.vmware.event.ItemCacheLoader --async-event-queue-id=item-writebehind-queue
```

Comment:
Why a REPLICATE region? We usually want users using a Partitioned region, especially when there are lots of writes happening.  Replicate is usually only used for data sets with a high amount of READs.